tests/integration_test/firebase_ui_database/database_list_view_test.dart

```diff
@@ -0,0 +1,224 @@
+// Copyright 2024, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:firebase_database/firebase_database.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:firebase_ui_database/firebase_ui_database.dart';
+import 'package:mockito/mockito.dart';
+import '../utils.dart';
+
+const _kTestPath = 'flutter-tests';
+
+void main() {
+  group('DatabaseListViewBuilder', () {
+    setUp(() async {
+      await clearReference(
+        rtdb.ref(_kTestPath),
+      );
+    });
+
+    testWidgets('Allows specifying custom error handler', (tester) async {
+      final builderSpy = ListViewBuilderSpy();
+      final ref = rtdb.ref('unknown');
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: FirebaseDatabaseListView(
```

Comment:
This is repeated often, it could be worth to add a tester.pumpApp function that would take query, errorBuilder and itemBuilder as parameters

Reply:
The low amount of repetition is probably not worth it 🦢

Reply:
> worth to add a tester.pumpApp function

might be just me, but it's easier to have "vanilla" experience. every bit of new functionality needs learning, this way it's very straightforward and code repetition doesn't matter much in tests anyways